### PR TITLE
Dataset example

### LINKED ISSUE #125: Multiple graphs dataset example
Spec doesn't have an example for running against a dataset.

From #111:

> As far as I can tell, all examples only have the default graph in the dataset. With quads, the graph is just one more endpoint of the edge, so probably no increased complexity as far as cases in the algorithm are concerned, but it looks a bit strange that it speaks about quads when throughout there are just triples in the examples (especially when serialised for hashing).

> I find myself wondering how hard the algorithm would have to determine that a [Möbius ladder graph](https://en.wikipedia.org/wiki/M%C3%B6bius_ladder) is not isomorphic to a [circular ladder graph](https://en.wikipedia.org/wiki/Ladder_graph#Circular_ladder_graph) of same order. (For graph isomorphism to appear as a subproblem of c14n, consider a dataset with two graphs whose names are blank nodes. At what point can it tell those nodes get different hashes?) Or for that matter, Möbius ladder to a circular ladder of order 2 more. The algorithm does not directly use multiplicity of hashes, does it? Only whether hashes are unique or not. Hence a 10-vertex 3-regular graph is not so obviously distinct from a 12-vertex 3-regular graph.

### PR DESCRIPTION
**Pull request by gkellogg**
* Add example of a dataset with a blank node named graph.
* Four dataset tests.
* Fix URL/NCname generation in HTML for tests and reports.

Fixes #125.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-canon/pull/129.html" title="Last updated on Jun 21, 2023, 2:31 PM UTC (0d8d955)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-canon/129/06f97b3...0d8d955.html" title="Last updated on Jun 21, 2023, 2:31 PM UTC (0d8d955)">Diff</a>

### CODE REVIEW

**Review by iherman — APPROVED** [review]
rdfjs-c14n passes the tests as described...

File: tests/rdfc10/test073-in.nq [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<http://example.org/test> <http://example.org/vocab#A> _:e0 .
+<http://example.org/test> <http://example.org/vocab#B> _:e0 .
+<http://example.org/test> <http://example.org/vocab#embed> _:e0 .
+<http://example.org/test> <http://example.org/vocab#graph> _:g1 .
+<http://example.org/test> <http://example.org/vocab#A> _:e0  _:g1 .
+<http://example.org/test> <http://example.org/vocab#B> _:e0 _:g1 .
+<http://example.org/test> <http://example.org/vocab#embed> _:e0 _:g1 .
+
+
```
Comment (yamdan) [inline tests/rdfc10/test073-in.nq:9]:
```suggestion
```

File: tests/rdfc10/test072-in.nq [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<http://example.org/test> <http://example.org/vocab#A> _:e0 .
+<http://example.org/test> <http://example.org/vocab#B> _:e0 .
+<http://example.org/test> <http://example.org/vocab#embed> _:e0 .
+<http://example.org/test> <http://example.org/vocab#A> _:e0 <http://example.org/g1> .
+<http://example.org/test> <http://example.org/vocab#B> _:e0 <http://example.org/g1> .
+<http://example.org/test> <http://example.org/vocab#embed> _:e0 <http://example.org/g1> .
+
```
Comment (yamdan) [inline tests/rdfc10/test072-in.nq:7]:
```suggestion
```

File: tests/rdfc10/test071-in.nq [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<http://example.org/test> <http://example.org/vocab#A> _:e0 .
+<http://example.org/test> <http://example.org/vocab#B> _:e0 .
+<http://example.org/test> <http://example.org/vocab#embed> _:e0 .
+<http://example.org/test> <http://example.org/vocab#A> _:e1 _:g1 .
+<http://example.org/test> <http://example.org/vocab#B> _:e1 _:g1 .
+<http://example.org/test> <http://example.org/vocab#embed> _:e1 _:g1 .
+
```
Comment (yamdan) [inline tests/rdfc10/test071-in.nq:7]:
```suggestion
```

**Review by yamdan — APPROVED** [review]
Thank you @gkellogg! My implementation also passes these additional tests.
I added a few suggestions for the example in index.html.

**Pull request merged**